src/components/ClientReservationListItem.js

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from "react";
+import { View, FlatList, RefreshControl } from "react-native";
+import { ListItem, ButtonGroup } from "react-native-elements";
+import { connect } from "react-redux";
+import { NavigationActions } from "react-navigation";
+import { MONTHS, DAYS, MAIN_COLOR } from "../constants";
+import { Spinner, EmptyList } from "./common";
+import { onClientReservationListRead } from "../actions";
+import moment from "moment";
+
+class ClientReservationListItem extends Component {
+  constructor(props) {
+    super(props);
+    props.onClientReservationListRead();
+    this.state = {
+      selectedIndex: 1,
+      filteredList: []
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.reservations !== this.props.reservations) {
+      this.updateIndex(this.state.selectedIndex);
+    }
+  }
+
+  OnPressItem = reservation => {
+    const navigateAction = NavigationActions.navigate({
+      routeName: "reservationsDetail",
+      params: reservation
+    });
+    this.props.navigation.navigate(navigateAction);
+  };
+
+  updateIndex = selectedIndex => {
+    this.setState({ selectedIndex });
+    const { reservations } = this.props;
+    var filteredList = [];
+    if (selectedIndex == 0) {
+      // turnos pasados
+      filteredList = reservations
+        .filter(res => res.endDate < moment() && res.state !== "Cancelado")
+        .sort((a, b) => a.startDate < b.startDate);
+    } else {
+      // turnos proximos
+      filteredList = reservations.filter(
+        res => res.startDate > moment() && res.state !== "Cancelado"
+      );
+    }
+    this.setState({ filteredList });
```

Comment:
`this.setState({ selectedIndex })` y `this.setState({ filteredList })` los estas ejecutando en dos lineas separadas. Eso hace que el componente se renderice dos veces, cuando podes evitarlo y hacer los dos seteos al ultimo:
`this.setState({ selectedIndex, filteredList })`

Reply:
buena observacion, yo tambnien tengo que cambiar eso en mi branch

Reply:
Muy bueno